lib/dust.js

```diff
@@ -861,6 +861,17 @@
     var body = bodies.block,
         skip = bodies['else'];
 
+    if (dust.isThenable(elem)) {
```

Comment:
The code was definitely redundant between the two methods.  I made a `mapThenable` method to address it.